packages/dht/src/connection/websocket/WebsocketConnectorRpcLocal.ts

```diff
@@ -1,292 +1,37 @@
-        })
-        const managedConnection = new ManagedConnection(this.localPeerDescriptor!, ConnectionType.WEBSOCKET_SERVER)
-        managedConnection.on('disconnected', () => this.ongoingConnectRequests.delete(keyFromPeerDescriptor(targetPeerDescriptor)))
-        managedConnection.setPeerDescriptor(targetPeerDescriptor)
-        this.ongoingConnectRequests.set(keyFromPeerDescriptor(targetPeerDescriptor), managedConnection)
-        return managedConnection
-    }
-
-    private onServerSocketHandshakeRequest(peerDescriptor: PeerDescriptor, serverWebsocket: IConnection) {
-
-        const peerId = peerIdFromPeerDescriptor(peerDescriptor)
-
-        if (this.ongoingConnectRequests.has(peerId.toKey())) {
-            const ongoingConnectReguest = this.ongoingConnectRequests.get(peerId.toKey())!
-            ongoingConnectReguest.attachImplementation(serverWebsocket)
-            ongoingConnectReguest.acceptHandshake()
-            this.ongoingConnectRequests.delete(peerId.toKey())
-        } else {
-            const managedConnection = new ManagedConnection(this.localPeerDescriptor!, ConnectionType.WEBSOCKET_SERVER, undefined, serverWebsocket)
-
-            managedConnection.setPeerDescriptor(peerDescriptor)
-
-            if (this.onIncomingConnection(managedConnection)) {
-                managedConnection.acceptHandshake()
-            } else {
-                managedConnection.rejectHandshake('Duplicate connection')
-                managedConnection.destroy()
-            }
-        }
-    }
-
-    public setLocalPeerDescriptor(localPeerDescriptor: PeerDescriptor): void {
-        this.localPeerDescriptor = localPeerDescriptor
-    }
-
-    public async destroy(): Promise<void> {
-        this.destroyed = true
-        this.rpcCommunicator.destroy()
-
-        const requests = Array.from(this.ongoingConnectRequests.values())
-        await Promise.allSettled(requests.map((conn) => conn.close('OTHER')))
-
-        const attempts = Array.from(this.connectingConnections.values())
-        await Promise.allSettled(attempts.map((conn) => conn.close('OTHER')))
-        this.connectivityChecker?.destroy()
-        await this.websocketServer?.stop()
-    }
-
-    // IWebsocketConnectorRpc implementation
     public async requestConnection(_request: WebsocketConnectionRequest, context: ServerCallContext): Promise<WebsocketConnectionResponse> {
         const senderPeerDescriptor = (context as DhtCallContext).incomingSourceDescriptor!
-        if (!this.destroyed && this.canConnectFunction(senderPeerDescriptor)) {
+        if (this.config.canConnect(senderPeerDescriptor)) {
             setImmediate(() => {
-                if (this.destroyed) {
-                    return
-                }
-                const connection = this.connect(senderPeerDescriptor)
-                this.onIncomingConnection(connection)
+                const connection = this.config.connect(senderPeerDescriptor)
```

Comment:
Would be good to have some sort of check here for cases where the connector is destroyed and the setImmediate block is executed. The setImmediate call might leak now if it is executed after stop and the timings are worst case